File: Difficulty: Easy/Square root of a number/square-root-of-a-number.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

  

// } Driver Code Ends
// Function to find square root
// x: element to find square root
class Solution{
  public:
    long long int floorSqrt(long long int x) 
    {
        // Your code goes here   
        
        long long int s=1;
        long long int e=x;
        long long ans=0;
       long long int mid=0;
        
        while(s<=e){
            mid=(s+e)/2;
            
            if(mid*mid==x){
                return mid;
            }
            else if(mid*mid>x){
                e=mid-1;
            }
            else {
                ans=mid;
                s=mid+1;
            }
        }
        return ans;
    }
};

//{ Driver Code Starts.

int main()
{
	int t;
	cin>>t;
	while(t--)
	{
		long long n;
		cin>>n;
		Solution obj;
		cout << obj.floorSqrt(n) << endl;
	}
    return 0;   
}

// } Driver Code Ends

File: Difficulty: Easy/Square root of a number/square-root-of-a-number.java
//{ Driver Code Starts
import java.util.Scanner;

class SquartRoot {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            long a = sc.nextInt();
            Solution obj = new Solution();
            System.out.println(obj.floorSqrt(a));
            t--;
        }
    }
}
// } Driver Code Ends


/*You are required to complete
this function*/

// Function to find square root
// x: element to find square root
class Solution {
    long floorSqrt(long n) {
        // Your code here
        return findSqrt(n);
    }
    long findSqrt(long n){
        long s=0;
        long  e=n;
        long mid=0;
        long ans=0;
        while(s<=e){
            
         mid=(s+e)/2;
            
            if(mid*mid==n){
                return mid;
            }
            
            if(mid * mid< n){
                ans=mid;
                s=mid+1;
            }
            else{
                e=mid-1;
            }
        }
        return ans;
    }
}
